services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -132,16 +136,45 @@ private boolean checkNoOverlapsInTimeWindow(List<TemporaryExposureKey> exposureK
 
       Integer[] sortedStartIntervalNumbers = exposureKeys.stream()
           .mapToInt(TemporaryExposureKey::getRollingStartIntervalNumber)
-          .sorted().boxed().toArray(Integer[]::new);
+          .distinct().sorted().boxed().toArray(Integer[]::new);
```

Comment:
This breaks the validation for TEK sets without flexible rolling periods, as it removed duplicate rolling start interval numbers. With some refactoring, `checkUniqueStartIntervalNumbers` could be removed altogether (see above).